fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/api/event/registry/RegistryAddObjectCallback.java

```diff
@@ -14,11 +14,23 @@
  * limitations under the License.
  */
 
-package net.fabricmc.fabric.impl.registry.callbacks;
+package net.fabricmc.fabric.api.event.registry;
 
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.impl.registry.ListenableRegistry;
 import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
 
 @FunctionalInterface
-public interface RegistryPostRegisterCallback<T> extends RegistryCallback<T> {
-	void onPostRegister(int rawId, Identifier id, T object);
+public interface RegistryAddObjectCallback<T> {
+	void onAddObject(int rawId, Identifier id, T object);
```

Comment:
There should be a matching RegistryReplaceObjectCallback

Reply:
We should not allow registry replacement. This was promoted by forge, but in fabric, we should ask modders to inject onto the code of the object they are modifying. That way, if two mods add hooks to leaves at the same time (changing two different properties, for example), they can be both applied, unlike in forge that only one is preserved.

Reply:
We don't always have a choice to efficiently inject into the code when a block or item is an instance of a shared class. Both options have their use cases.

Reply:
There are good reasons to replace, and there are some good reasons not to. I don’t see any reason I’m not allowing it, it’s up to the mod to try and be as compatible as possible.

Reply:
Sure. A warning like that in forge imo is the best.

Reply:
I'm talking about the *callback*.

Though, to minimize the amount of work on mods' side, we should probably just emit Remove, then Add.

Reply:
Handling such a situation does not imply supporting such a situation - if another loader below performs a replacement, you'd still want to know, no?

Reply:
Done.